src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/CryptoUtils.java

```diff
@@ -0,0 +1,334 @@
+package com.ironcorelabs.tenantsecurity.kms.v1;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PushbackInputStream;
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import java.util.concurrent.CompletableFuture;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import com.ironcorelabs.proto.DocumentHeader;
+import com.ironcorelabs.tenantsecurity.utils.CompletableFutures;
+
+class CryptoUtils {
+    static final String AES_ALGO = "AES/GCM/NoPadding";
+    public static final int IV_BYTE_LENGTH = 12;
+    static final int GCM_TAG_BIT_LENGTH = 128;
+    // the size of the fixed length portion of the header (version, magic, size)
+    static final int DOCUMENT_HEADER_META_LENGTH = 7;
+    static final byte CURRENT_DOCUMENT_HEADER_VERSION = 3;
+    static final int GCM_TAG_BYTE_LEN = GCM_TAG_BIT_LENGTH / 8;
+
+    static final byte[] DOCUMENT_MAGIC = { 73, 82, 79, 78 }; // bytes for ASCII IRON
+                                                             // characters
+
+    static final int HEADER_META_LENGTH_LENGTH = 2;
+    // The number of fixed size bytes of header at the front of all CMK documents.
+    // After this length is the probobuf-encoded header bytes, which
+    // might be empty.
+    static final int HEADER_FIXED_SIZE_CONTENT_LENGTH = 1 + DOCUMENT_MAGIC.length + HEADER_META_LENGTH_LENGTH;
+    // How many bytes to read in each loop when doing streaming encryption.
+    static final int STREAM_CHUNKING = 32;
+    static final int MAX_HEADER_SIZE = 65535; // 256 * 255 + 255 since we do a 2 byte size.
```

Comment:
aka "2^16 - 1" :)

Reply:
1) The comment came from somewhere else
2) I think the math makes more sense to the lay person reading it

(Secretly I wanted to change it...)